packages/expo-updates/ios/EXUpdates/AppLoader/RemoteAppLoader.swift

```diff
@@ -41,52 +41,73 @@ public final class RemoteAppLoader: AppLoader {
         successBlockArg(updateResponse)
         return
       }
-      // even if update is nil (meaning we didn't load a new update),
-      // we want to persist the header data from remoteUpdateResponse
-      if let remoteUpdateResponse = strongSelf.remoteUpdateResponse,
-        let responseHeaderData = remoteUpdateResponse.responseHeaderData {
-        strongSelf.database.databaseQueue.async {
-          do {
-            try strongSelf.database.setMetadata(withResponseHeaderData: responseHeaderData, scopeKey: strongSelf.config.scopeKey)
-            successBlockArg(updateResponse)
-          } catch {
-            let cause = UpdatesError.remoteAppLoaderHeaderDataError(cause: error)
-            strongSelf.logger.error(cause: cause, code: UpdatesErrorCode.unknown)
-            errorBlockArg(cause)
-          }
+      strongSelf.processUpdateResponse(updateResponse, success: successBlockArg, error: errorBlockArg)
```

Comment:
```suggestion
       guard let self else {
        successBlockArg(updateResponse)
        return
      }
      processUpdateResponse(updateResponse, success: successBlockArg, error: errorBlockArg)
```
I know you didn't add this but lets clean it up 😅